quickwit/quickwit-metastore/src/metastore/file_backed_metastore/file_backed_index/serialize.rs

```diff
@@ -57,7 +57,7 @@ pub(crate) struct FileBackedIndexV0_6 {
     #[serde(rename = "index")]
     metadata: IndexMetadata,
     splits: Vec<Split>,
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
```

Comment:
I agree with you and will remove it eventually. I want to hide the shards stuff until it's stable for now.